### ISSUE TITLE
conda-defaults and conda-forge binary incompatible

### ISSUE THREAD

**Issue by sebastian-luna-valero** [bioconda/bioconda-utils#132]
Following up on:
https://github.com/ContinuumIO/anaconda-issues/issues/1403

Copy and paste:

> The only possible resolution, while conda-defaults and coda-forge remain binary incompatible is not to mix them in the same environment

Should `conda-forge` still be encouraged as part of the setup for bioconda in the documentation?
http://bioconda.github.io/contribute-a-recipe.html#use-your-new-recipe

Best regards,
Sebastian

**Comment by johanneskoester** [comment]
Not yet. We are in the process of migrating but currently, a lot of packages (esp. R) are still built against defaults. We hope to finish the migration as soon as possible though.

**Comment by johanneskoester** [comment]
In fact, mixing is fine when defaults has a higher priority than conda forge. This is what we require in the docs.

**Comment by sebastian-luna-valero** [comment]
Thanks!

**Comment by sebastian-luna-valero** [comment]
Hello,

Are there any docs/policy about the migration of packages from `bioconda` to `conda-forge`?

I am in the process of adding the [libvips](https://github.com/jcupitt/libvips/issues/673) package, and was wondering whether it is better to add it to `bioconda` or `conda-forge` directly.

Best regards,
Sebastian

**Comment by sebastian-luna-valero** [comment]
Also: `libvips` depends on `openslide` and `graphicsmagick` which are in `bioconda` and not in `conda-forge`

**Comment by bgruening** [comment]
If you have the exactly same package in conda-forge we can simply remove it from bioconda. We would like to move rather sooner than later all general purpose packages to conda-forge.

**Comment by sebastian-luna-valero** [comment]
Thanks @bgruening 

Do [libvips](https://jcupitt.github.io/libvips/), [openslide](http://openslide.org/) and [graphicsmagick](http://www.graphicsmagick.org/) fall into the `conda-forge` category? If so, I will upload them all to `conda-forge` and then we'll have to remove the last two from `bioconda`.

**Comment by johanneskoester** [comment]
I would say that openslide is bioconda (because it is a patology tool, right?). Whereas libvips and graphicsmagick are clearly conda-forge. Is it feasible for you to add those to conda-forge?

**Comment by sebastian-luna-valero** [comment]
I am working on libvips [here](https://github.com/jcupitt/libvips/issues/673). I have also asked `conda-forge` people about dependencies in `bioconda` [here](https://github.com/conda-forge/staged-recipes/issues/3144). Apparently, all of them have to be in `conda-forge`, that's why I am asking before going ahead.

**Comment by ThomasWollmann** [comment]
Openslide package maintainer here. Yes, openslide is biology/medicine related. IMHO it should stay at bioconda. However, graphicsmagick is a general purpose library, so we can move it to conda-forge.

**Comment by sebastian-luna-valero** [comment]
Thanks All!

Sorry this might be a stupid question. How do I go about uploading libvips to `conda-forge`, if I want `openslide` to be a dependency in `bioconda`?

Please see conda-forge's answer [here](https://github.com/conda-forge/staged-recipes/issues/3144)

**Comment by bgruening** [comment]
@sebastian-luna-valero this is not possible. Either we move all of this to conda-forge or we leave it here :(

**Comment by johanneskoester** [comment]
It should be fine to have libvips here then. If it depends on a biology/medicine package, I guess we can assume that it is only relevant for our community.

**Comment by sebastian-luna-valero** [comment]
Will do, many thanks!

**Comment by jcupitt** [comment]
Hello, I'm the libvips maintainer. We've struggled with this issue (what category to put libvips in) in [homebrew](https://brew.sh/) as well. 

libvips started out as a scientific image processing library (that's what I use it for) and that's why it supports things lie openslide, libmatio (for Matlab files), and cfitsio (for FITS astronomy images) as input methods. In the last five years or so it's become popular for server-side image processing, since it's ~4x faster than graphicsmagick and uses ~10x less memory. It's one of the most-used image processing libraries in node.js, and it's (I think) the most popular engine for resizing image proxies.  

As a result, it's been moved from homebrew-science into homebrew-core and had the -science dependencies removed. If you want libvips with openslide (for example), you now need to pass a set of extra flags during `brew install`.  Longer term, the plan is to have a second package called libvips-science with the science dependencies enabled. 

tldr: you could consider having two libvips packages. Does conda support sub-packages?

**Comment by johanneskoester** [comment]
That would work. We can have a conda-forge version that is compiled without openslide support. There is no sub-package concept in conda, but this is fine since bioconda and conda-forge are separate channels and it is possible to specify their priority. If people use bioconda, they will get the science version. If they only use conda-forge, they would get the non-science version.

Btw. congrats to the huge success of your tool!
